### ISSUE TITLE
Installation issue: kokkos-kernels@3.4.01 with rocm

### ISSUE THREAD

**Issue by balay** [LLNL/spack#27684]
### Steps to reproduce the issue

```console
spack install kokkos-kernels ^kokkos+rocm amdgpu_target=gfx908
```

### Information on your system

```console

$ spack debug report
* **Spack:** 0.17.0-316-8ff81d4
* **Python:** 3.6.8
* **Platform:** linux-centos7-broadwell
* **Concretizer:** clingo
```

### Additional information

```
g++: error: unrecognized command-line option '-fno-gpu-rdc'
g++: error: unrecognized command-line option '--amdgpu-target=gfx908'
g++: error: unrecognized command-line option '--amdgpu-target=gfx908'
make[2]: *** [src/CMakeFiles/kokkoskernels.dir/impl/generated_specializations_cpp/abs/Blas1_abs_eti_DOUBLE_LAYOUTLEFT_EXECSPACE_SERIAL_MEMSPACE_HOSTSPACE.cpp.o] Error 1

```

[spack-build-out.txt](https://github.com/spack/spack/files/7615117/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/7615118/spack-build-env.txt)

cc: @brian-kelley  @lucbv  @srajama1

BTW: Should kokkos-kernels support a `rocm` variant?

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by balay** [comment]
This fix gets the build going for me.
```patch
diff --git a/var/spack/repos/builtin/packages/kokkos-kernels/package.py b/var/spack/repos/builtin/packages/kokkos-kernels/package.py
index c41a1607aa..8b046dd89f 100644
--- a/var/spack/repos/builtin/packages/kokkos-kernels/package.py
+++ b/var/spack/repos/builtin/packages/kokkos-kernels/package.py
@@ -102,8 +102,11 @@ def cmake_args(self):
             options.append("-DSpack_WORKAROUND=On")
 
         options.append("-DKokkos_ROOT=%s" % spec["kokkos"].prefix)
-        # Compiler weirdness due to nvcc_wrapper
-        options.append("-DCMAKE_CXX_COMPILER=%s" % spec["kokkos"].kokkos_cxx)
+        if spec.satisfies('^kokkos+rocm'):
+            options.append("-DCMAKE_CXX_COMPILER=%s" % spec['hip'].hipcc)
+        else:
+            # Compiler weirdness due to nvcc_wrapper
+            options.append("-DCMAKE_CXX_COMPILER=%s" % spec["kokkos"].kokkos_cxx)
 
         if self.run_tests:
             options.append("-DKokkosKernels_ENABLE_TESTS=ON")
```

**Comment by lucbv** [comment]
Hi @balay 
I do not recall having issues last time I did such a build on Spock but I will try to replicate this issue.
In any case we also released version 3.5.0 of Kokkos and Kokkos-Kernels so I will also update the version appropriately.

**Comment by balay** [comment]
@lucbv This build is on a linux box (at u oregon) - and not spock. I'm yet to try this on spock.

**Comment by balay** [comment]
@lucbv this change also works for me on spock - so  including it in my PR at https://github.com/spack/spack/pull/27779

**Comment by balay** [comment]
Aditional note: without this fix - kokkos-kernels build also fails on spock for me

[spack-build-out.txt](https://github.com/spack/spack/files/7651260/spack-build-out.txt)

**Comment by balay** [comment]
@keitat @lucbv Could you approve this fix at #27779 ?
